src/main/java/com/altenheim/kalender/controller/viewController/SettingsViewController.java

```diff
@@ -1,23 +1,82 @@
 package com.altenheim.kalender.controller.viewController;
 
-import com.altenheim.kalender.interfaces.ISettingsController;
 import com.altenheim.kalender.models.SettingsModel;
 
-public class SettingsViewController extends ResponsiveController implements ISettingsController
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.TextField;
+
+public class SettingsViewController extends ResponsiveController
 { 
-    private ISettingsController settingsController;
     private SettingsModel settings;
+    @FXML
+    private TextField Studiengang;
+
+    @FXML
+    private TextField Kurs;
 
-    public SettingsViewController(ISettingsController settingsController, SettingsModel settings)
+    @FXML
+    void save(ActionEvent event) {
+        String userinput = urlbuilder(Studiengang.getText(),Kurs.getText());
+         settings.seturlvariables(userinput);
+    }
+    public SettingsViewController(SettingsModel settings)
     {
-        this.settingsController = settingsController;
         this.settings = settings;
     }
     
     public void changeContentPosition() 
     {
         
     }    
+    
+
+        public String urlbuilder(String inputa, String inputb) {
+            
+            String semester = "";
+            String kurs = "";
+            
+            
+            if (inputb.contains("1")) {
+                 semester = "1";	
+                }
+            else if (inputb.contains("2")) {
+                 semester = "2";	
+                }
+            else if (inputb.contains("3")) {
+                 semester = "3";	
+                }
+            else if (inputb.contains("4")) {
+                 semester = "4";	
+                }
+            else if (inputb.contains("5")) {
+                 semester = "5";	
+                }
+            else if (inputb.contains("6")) {
+                 semester = "6";	
+                }
+            else {
+            System.out.println("Fehler bei Eingabe des Semesters");
+            }
```

Comment:
Refactor: Zeile 40 bis 60 (eine for schleife mit einem if und ein if für Falsche Eingabe max)